runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/graph/CreateExecutableStageNodeFunction.java

```diff
@@ -167,19 +167,24 @@ public Node apply(MutableNetwork<Node, Edge> input) {
       componentsBuilder.putEnvironments(envId, Environments.JAVA_SDK_HARNESS_ENVIRONMENT);
     }
 
-    // Use default WindowingStrategy as the fake one. But it may be overrided later if it's a FixedWindow.
+    // By default, use GlobalWindow for all languages.
+		// For java, if there is a IntervalWindowCoder, then use FixedWindow instead.
     // TODO: should get real WindowingStategy from pipeline proto.
-    String fakeWindowingStrategyId = "fakeWindowingStrategy" + idGenerator.getId();
+    String fakeGlobalWindowingStrategyId = "fakeGlobalWindowingStrategy" + idGenerator.getId();
+    String fakeFixedWindowingStrategyId = "fakeFixedWindowingStrategy" + idGenerator.getId();
```

Comment:
You could call this intervalWindowEncodingWindowingStrategyId or similar to clearly identify the properties you need to leverage here.